Tests/Common/IntegrationTestFixtureBase.cs

```diff
@@ -322,7 +322,10 @@ public async Task UpdateExistingConcentratorCrcValues(StationEui stationEui, uin
                                                crc,
                                                crc,
                                                deserializedCupsTwinInfo.CupsCredentialUrl,
-                                               deserializedCupsTwinInfo.TcCredentialUrl);
+                                               deserializedCupsTwinInfo.TcCredentialUrl,
```

Comment:
If you update `CupsTwinInfo` as follows so that its `CupsCredCrc` and `TcCredCrc` properties have initialisers:

```diff
diff --git a/LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/CupsTwinInfo.cs b/LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/CupsTwinInfo.cs
index cb7959e7..774c3649 100644
--- a/LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/CupsTwinInfo.cs
+++ b/LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/CupsTwinInfo.cs
@@ -44,11 +44,11 @@ namespace LoRaWan.NetworkServer.BasicsStation
 
         [JsonPropertyName("cupsCredCrc")]
         [JsonProperty("cupsCredCrc")]
-        public uint CupsCredCrc { get; }
+        public uint CupsCredCrc { get; init; }
 
         [JsonPropertyName("tcCredCrc")]
         [JsonProperty("tcCredCrc")]
-        public uint TcCredCrc { get; }
+        public uint TcCredCrc { get; init; }
 
         [JsonPropertyName("cupsCredentialUrl")]
         [JsonProperty("cupsCredentialUrl")]
```

then you can use `with` here to have the compiler use the copy constructor (which is what you're doing in effect here):

```c#
            var newCupsInfo = JsonConvert.DeserializeObject<CupsTwinInfo>(cupsJson) with
            {
                CupsCredCrc = crc,
                TcCredCrc = crc
            };
```

Now this code doesn't have to be updated as more properties are added/removed in `CupsTwinInfo` or its constructor updated.

Reply:
Done with 48377fa025c6d2ec6185c030b5bcd6a13f89d4f2.